postgresql-plugin/src/main/java/io/cdap/plugin/postgres/PostgresDBRecord.java

```diff
@@ -49,24 +52,44 @@ public PostgresDBRecord() {
   @Override
   protected void handleField(ResultSet resultSet, StructuredRecord.Builder recordBuilder, Schema.Field field,
                              int columnIndex, int sqlType, int sqlPrecision, int sqlScale) throws SQLException {
+    ResultSetMetaData metadata = resultSet.getMetaData();
     if (isUseSchema(resultSet.getMetaData(), columnIndex)) {
       setFieldAccordingToSchema(resultSet, recordBuilder, field, columnIndex);
     } else {
+      int columnType = metadata.getColumnType(columnIndex);
+      if (columnType == Types.NUMERIC) {
+        Schema nonNullableSchema = field.getSchema().isNullable() ?
+          field.getSchema().getNonNullable() : field.getSchema();
+        int precision = metadata.getPrecision(columnIndex);
+        if (precision == 0 && Schema.Type.STRING.equals(nonNullableSchema.getType())) {
+          // When output schema is set to String for precision less numbers
+          recordBuilder.set(field.getName(), resultSet.getString(columnIndex));
+        } else if (Schema.LogicalType.DECIMAL.equals(nonNullableSchema.getLogicalType())) {
+          BigDecimal orgValue = resultSet.getBigDecimal(columnIndex);
+          if (orgValue != null) {
+            BigDecimal decimalValue = new BigDecimal(orgValue.toPlainString())
+              .setScale(nonNullableSchema.getScale(), RoundingMode.HALF_EVEN);
```

Comment:
this rounding behavior seems like it should be documented

Reply:
Documented in the same document.